help_function/src/util_group_load_store_test.cpp

```diff
@@ -0,0 +1,199 @@
+// ====------ onedpl_test_group_load.cpp------------ *- C++ -* ----===//
+
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//
+// ===----------------------------------------------------------------------===//
+#include <dpct/dpct.hpp>
+#include <dpct/dpl_utils.hpp>
+#include <iostream>
+#include <oneapi/dpl/iterator>
+#include <sycl/sycl.hpp>
+
+template <dpct::group::load_algorithm T>
+bool helper_validation_function(const int *ptr, const char *func_name) {
+  if constexpr (T == dpct::group::load_algorithm::BLOCK_LOAD_DIRECT) {

```

Comment:
It seems like the only difference between verification for "direct" and "striped" load algorithms is the name.  This makes sense because we should "unmap" any "map" via the store operation as long as they match.

However, it doesn't make sense to just have a branch and repeat the code.  You can branch for the name, and run the same verification, or just embed the info you want in `func_name`.

Reply:
Yes that makes sense, thanks